src/NaturalLanguage/NLLanguageRecognizer.cs

```diff
@@ -34,10 +34,11 @@ public static NLLanguage GetDominantLanguage (string @string)
 			var nsstring = NSString.CreateNative (@string);
 			var nslang = _GetDominantLanguage (nsstring);
 			var lang =  NLLanguage.Undetermined;
-			if (nslang != null)
+			if (nslang != null) {
 				lang = NLLanguageExtensions.GetValue (nslang);
+				nslang.Dispose ();
+			}
```

Comment:
Change does fix the issue, what about making the code smaller and probably easier to read. GetValue is generated as :

```csharp
public static NLLanguage GetValue (NSString constant)
                {
                        if (constant == null)
                                return NLLanguage.Unevaluated;
                        if (constant.IsEqualTo (NLLanguageUndetermined))
                                return NLLanguage.Undetermined;
                        if (constant.IsEqualTo (NLLanguageAmharic))
                                return NLLanguage.Amharic;
                        if (constant.IsEqualTo (NLLanguageArabic))
                                return NLLanguage.Arabic;
                        if (constant.IsEqualTo (NLLanguageArmenian))
                                return NLLanguage.Armenian;
                        if (constant.IsEqualTo (NLLanguageBengali))
                                return NLLanguage.Bengali;
                        if (constant.IsEqualTo (NLLanguageBulgarian))
                                return NLLanguage.Bulgarian;
                        if (constant.IsEqualTo (NLLanguageBurmese))
                                return NLLanguage.Burmese;
                        if (constant.IsEqualTo (NLLanguageCatalan))
                                return NLLanguage.Catalan;
                        if (constant.IsEqualTo (NLLanguageCherokee))
                                return NLLanguage.Cherokee;
...
                        return NLLanguage.Unevaluated;
}
```
So, we can pass null to GetValue, in that case we could do something like:
```csharp
var lang = NLLanguageExtensions.GetValue (nslang); // if null or not known, returns NLLanguage.Unevaluated
nslang?.Dispose();
```

We remove the need to init the lang to NLLanguage.Undetermined which might not be the correctly expected result  (at least from an objc user perspective) and we remove the if branching.

Could we get a test case added here: https://github.com/xamarin/xamarin-macios/blob/master/tests/monotouch-test/NaturalLanguage/NLLanguageRecognizerTest.cs

PS: THANKS for the PR!!!

Reply:
changes request addressed